apps/ledger-live-desktop/src/newArch/WalletSync/Flows/Activation/03-ActivationOrSynchroWithTrustchain.tsx

```diff
@@ -2,26 +2,52 @@ import React, { useCallback, useEffect } from "react";
 import Loading from "../../components/LoadingStep";
 import { useTranslation } from "react-i18next";
 import { UnsecuredError } from "./03-UnsecuredError";
+import { setMemberCredentials, setTrustchain } from "@ledgerhq/trustchain/store";
+import { useDispatch } from "react-redux";
+import { Device } from "@ledgerhq/live-common/hw/actions/types";
+import { Flow, Step } from "~/renderer/reducers/walletSync";
+import { setFlow } from "~/renderer/actions/walletSync";
+import { runWithDevice, useTrustchainSdk } from "../../useTrustchainSdk";
 
 type Props = {
-  goNext: () => void;
+  device: Device | null;
 };
 
-export default function ActivationOrSynchroWithTrustchain({ goNext }: Props) {
+export default function ActivationOrSynchroWithTrustchain({ device }: Props) {
   const { t } = useTranslation();
+  const sdk = useTrustchainSdk();
+  const dispatch = useDispatch();
 
   const hasError = false;
 
-  // TO CHANGE WHEN INTRAGRATION WITH TRUSTCHAIN
-  const stuffHandledByTrustchain = useCallback(() => {
-    goNext();
-  }, [goNext]);
-  // TO CHANGE WHEN INTRAGRATION WITH TRUSTCHAIN
+  const stuffHandledByTrustchain = useCallback(async () => {
+    const memberCredentials = await sdk.initMemberCredentials();
+    dispatch(setMemberCredentials(memberCredentials));
+
+    const seedIdToken = await runWithDevice(device?.deviceId, transport =>
+      sdk.authWithDevice(transport),
+    );
+    const { trustchain, hasCreatedTrustchain } = await runWithDevice(
+      device?.deviceId,
+      transport => {
+        return sdk.getOrCreateTrustchain(transport, seedIdToken, memberCredentials);
+      },
+    );
+    dispatch(setTrustchain(trustchain));
+
+    console.log("trustchain", trustchain);
+    dispatch(
+      setFlow({
+        flow: Flow.Activation,
+        step: hasCreatedTrustchain ? Step.ActivationFinal : Step.SynchronizationFinal,
+      }),
+    );
+  }, [sdk, dispatch, device?.deviceId]);
+
   useEffect(() => {
-    setTimeout(() => {
-      !hasError && stuffHandledByTrustchain();
-    }, 3000);
-  }, [hasError, stuffHandledByTrustchain]);
+    !hasError && stuffHandledByTrustchain();
```

Comment:
improvements i see:
- we need to handle possible error thrown by stuffHandledByTrustchain and display them to user (using the Generic Error rendering which should tell the user the precise error instead of the UnsecuredError )
- Q: is there a way, as a user, to interrupt the flow? because we don't handle the unsubscribe of these, and it's possible that runWithDevice never ends which can introduce race conditions